pymc_marketing/mmm/base.py

```diff
@@ -472,8 +474,136 @@ def compute_channel_contribution_original_scale(self) -> DataArray:
+        elbow_y = michaelis_menten(k, L, k)
+
+        if x_stop is not None:
+            x_limit = x_stop
+        else:
+            x_limit = k * (0.99 * L / (L * 0.01))
+
+        x_fit = np.linspace(0, x_limit, 1000)
+        y_fit = michaelis_menten(x_fit, L, k)
+
+        ax.plot(x_fit, y_fit, color=f"C{color_index}", label="Fit Curve", alpha=0.6)
+        ax.plot(
+            k,
+            elbow_y,
+            "go",
+            color=f"C{color_index}",
+            markerfacecolor="white",
+        )
+
+        ax.set(xlabel="Spent", ylabel="Contribution")
+        ax.legend()
+
+    def optimize_channel_budget_for_maximum_contribution(
+        self,
+        total_budget: int,
+        *,
+        parameters: Dict[str, Tuple[float, float]],
+        budget_bounds: Optional[Dict[str, Tuple[float, float]]],
+    ) -> pd.DataFrame:
+        """
+        Optimize the allocation of a given total budget across multiple channels to maximize the expected contribution.
+
+        The optimization is based on the Michaelis-Menten equation, where each channel's contribution
+        follows a saturating function of its allocated budget. The function seeks the budget allocation
+        that maximizes the total expected contribution across all channels.
+
+        Parameters
+        ----------
+        total_budget : int, requiere
+            The total budget to be distributed across channels.
+        parameters : dict, requiere
+            A dictionary where keys are channel names and values are tuples (L, k) representing the
+            Michaelis-Menten parameters for each channel.
+        budget_bounds : dict, optional
+            An optional dictionary defining the minimum and maximum budget for each channel.
+            If not provided, the budget for each channel is constrained between 0 and its L value.
+
+        Returns
+        -------
+        DataFrame
+            A pandas DataFrame containing the allocated budget and contribution information.
+
+        Raises
+        ------
+        ValueError
+            If any of the required parameters are not provided or have an incorrect type.
+        """
+        if not isinstance(budget_bounds, dict):
+            raise ValueError("The 'budget_bounds' parameter must be a dictionary.")
+
+        if not isinstance(total_budget, (int, float)):
+            raise ValueError(
+                "The 'total_budget' parameter must be an integer or float."
+            )
+
+        if not parameters:
+            raise ValueError(
+                "The 'parameters' argument (keyword-only) must be provided and non-empty."
+            )
+
+        return budget_allocator(
+            total_budget=total_budget,
+            channels=list(self.channel_columns),
+            parameters=parameters,
+            budget_ranges=budget_bounds,
+        )
+
+    def compute_channel_estimate_points_original_scale(self) -> Dict:
```

Comment:
I would rename this method as `compute_channel_plateat_points_original_scale`

Reply:
I switch to `compute_channel_curve_parameters_original_scale` given the fact we can use a differents curve fit now. What do you think?